assets/js/app.js

```diff
@@ -0,0 +1,59 @@
+let products = [];
+
+function collectData() {
+  const productName = document.getElementById(`productName`).value;
+  const productPrice = document.getElementById(`productPrice`).value;
+  const category = document.getElementById(`category`).value;
+  const url = document.getElementById(`url`).value;
+
+  return {
+    productName: productName,
+    productPrice: productPrice,
+    category: category,
+    url: url,
+  };
+}
+
+function generateHTML(data, index) {
+  const newHTML = `
+  <tr id="row-${index}">
+  <td>${data.productName}</td>
+  <td>${data.productPrice} ₪</td>
+  <td>${data.category}</td>
+  <td><img src="${data.url}" width="100" height="100" /></td>
+  <td><button onclick="remove(${index})" class="btn btn-danger">Remove</button></td>
+</tr>`;
+  return newHTML;
+}
+
+function renderHTML() {
+  const tableBody = document.getElementById("tableBody");
+  tableBody.innerHTML = "";
+
+  for (let i = 0; i < products.length; i++) {
+    const newHTML = generateHTML(products[i], i);
+    tableBody.innerHTML += newHTML;
```

Comment:
you should not update the DOM in a loop as it is not efficient. the browser has to re-render with each iteration. you should create a variable to cumulate the dhtml and set it to the DOM in a single command, once